test/test_utils/sagemaker.py

```diff
@@ -332,17 +333,20 @@ def execute_local_tests(image, pytest_cache_params):
                     output = subprocess.check_output(f"cat test/{job_type}_{tag}_sm_local.xml", shell=True,
                                                      executable="/bin/bash")
                     pytest_cache_util.upload_pytest_cache_from_ec2_to_s3(ec2_conn_new, path, **pytest_cache_params)
-                    if 'failures="0"' not in str(output):
+                    if 'failures="0"' not in str(output) and not is_nightly_context():
                         raise ValueError(f"Sagemaker Local tests failed for {image}")
             else:
-                ec2_conn.run(pytest_command)
-                print(f"Downloading Test reports for image: {image}")
-                ec2_conn.get(ec2_test_report_path, os.path.join("test", f"{job_type}_{tag}_sm_local.xml"))
-    except:
-        print(f"Exception {sys.exc_info()[0]} occurred")
+                try:
+                    ec2_conn.run(pytest_command)
+                    print(f"Downloading Test reports for image: {image}")
+                    ec2_conn.get(ec2_test_report_path, os.path.join("test", f"{job_type}_{tag}_sm_local.xml"))
+                except:
+                    if not is_nightly_context():
+                        raise ValueError(f"Sagemaker Local tests failed for {image}")
```

Comment:
Please raise the same exception as the one from the `try` block, rather than raising a new `ValueError`, because this breaks the flow of the Traceback.

For example:
```
try:
    ...
except Exception as e:
    if is_nightly_context():
        LOGGER.warning(f"SageMaker Local tests failed for {image} with exception: {e}")
    else:
        raise e
```

Reply:
Also, please add the `except:` block back into the outer `try-except-finally` block.

```
            else:
                try:
                    ec2_conn.run(pytest_command)
                    ...
                except Exception as e:
                    ....
    except:
        print(f"Exception {sys.exc_info()[0]} occurred")
```

Reply:
Do you want to bring in the logging functionality into this module? Up until now there was no logging mechanism.